### ISSUE TITLE
[Feature]: Email links - Base URL Setting

### ISSUE THREAD

**Issue by Er1c85** [Leantime/leantime#2848]
### What is your set up?

Cloud Hosted

### Is your feature request related to a problem? Please describe

When Leantime sends emails without a `LEAN_APP_URL` defined, it has been using a local DNS name when generating the links.

### Describe the solution you'd like

A config option (or better yet, a setting on the web interface) to support configuring a BASE_URL for any email links.

**Comment by marcelfolaron** [comment]
Does this issue get resolved when LEAN_APP_URL is set?

**Comment by Er1c85** [comment]
> Does this issue get resolved when LEAN_APP_URL is set?

Yes because with LEAN_APP_URL set, the ability to run Leantime from other BASE_URLs no longer exists.  The configuration I was using was such that I would often use a VPN and local DNS address, but I still had a internet-facing URL for external users that don't have VPN access.  Whichever user is active when the email message is generated by the server, it is their BASE_URL that is used for links in the email message.

My suggestion to keep all functionality the same and fix this issue would be to, when users connect, store their specific BASE_URL in their user settings database.  Then, when that user is messaged, use the BASE_URL from their user settings to issue them an email message that references the server the same way they had been when last connected.

Thanks again for your support!
